ui/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessRepository.scala

```diff
@@ -32,26 +32,27 @@ trait ProcessRepository[F[_]] extends Repository[F] with EspTables {
       .sortBy(_.createDate.desc)
 
   protected def fetchLastDeployedActionPerProcessQuery: Query[(api.Rep[Long], (ProcessActionEntityFactory#ProcessActionEntity, api.Rep[Option[CommentEntityFactory#CommentEntity]])), (Long, (ProcessActionEntityData, Option[CommentEntityData])), Seq] =
-    fetchLastActionPerProcessQuery.filter(_._2._1.action === ProcessActionType.Deploy)
+    fetchLastActionPerProcessQuery
+      .filter(_._2._1.action === ProcessActionType.Deploy)
 
   protected def fetchLastActionPerProcessQuery: Query[(Rep[Long], (ProcessActionEntityFactory#ProcessActionEntity, Rep[Option[CommentEntityFactory#CommentEntity]])), (Long, (ProcessActionEntityData, Option[CommentEntityData])), Seq] =
     processActionsTable
       .groupBy(_.processId)
       .map { case (processId, group) => (processId, group.map(_.performedAt).max) }
       .join(processActionsTable)
-      .on { case ((processId, latestDeployedAt), deployAction) => deployAction.processId === processId && deployAction.performedAt === latestDeployedAt } //We fetch exactly this one  with max deployment
-      .map { case ((processId, _), deployAction) => processId -> deployAction }
+      .on { case ((processId, maxPerformedAt), action) => action.processId === processId && action.performedAt === maxPerformedAt } //We fetch exactly this one  with max deployment
+      .map { case ((processId, _), action) => processId -> action }
```

Comment:
Why in this query we don't need to sort? Also IMO names of those query methods are confusing. Those queries return sequences, not one action. We should either change those queries to return first element or change name that will be more descriptive that we fetch actions (not last) per process in desc order.

Reply:
This query is something like big cache for last action for each process (there is filtering by max date - latest). We use it at process list - in that way we create only one query for last action for each process instead query per process.